language-plutus-core/test/types/recursiveWrap.plc.golden

```diff
@@ -0,0 +1,18 @@
+Type mismatch at 3:6 in term
+  '(abs
+    r
+    (type)
+    (lam
+      z
+      r
+      (lam
+        f
+        (fun a (fun (fix list (all r (type) (fun r (fun (fun a (fun list r)) r)))) r))
+        z
+      )
+    )
+  )'.
+Expected type
+  '(all r (type) (fun r (fun (fun a (fun (fix list (all r (type) (fun r (fun (fun a (fun list r)) r)))) r)) r)))',
+found type
+  '(all r (type) (fun r (fun (fun a (fun (fix list (all r (type) (fun r (fun (fun a (fun list r)) r)))) r)) r)))'
```

Comment:
It looks like type equality or pretty-printing needs to be fixed. Should I do this in a separate PR?

Reply:
I've headed home for the evening but yeah, this is where I'd got stuck - it does look rather suspicious!

Reply:
Sounds good. I will try to fix this. I'm not sure if pretty-printing ran awry or equality :\

Reply:
You'll need the renamer fix too.